### PR TITLE
support for Stream ID re-use

### PR DESCRIPTION
**Pull request by benjchristensen**
Possible wording if we agree to allow Stream ID re-use. Discussion in https://github.com/ReactiveSocket/reactivesocket/issues/221

### CODE REVIEW

**Comment by benjchristensen** [comment]
@tmontgomery Does that wording change improve things and address the interoperability concern?

**Comment by benjchristensen** [comment]
Thanks @stevegury for the review.

**Comment by benjchristensen** [comment]
Merging based on approvals of @tmontgomery and @stevegury.

Todd, if you have wording changes, please let me know, otherwise I think my changes should address your concern.

**Comment by tmontgomery** [comment]
I'm good.

File: Protocol.md [hunk 1]
```diff
@@ -216,7 +216,22 @@ Stream IDs on the server MUST start at 2 and increment by 2 sequentially, such a
...
+
+1) If Stream ID re-use is not employed:
+
+- No new streams can be created, thus a new connection MUST be established to create new streams once the max has been met. 
+
+2) If Stream IDs are re-used:
+
+- The Requester MUST re-use IDs by wrapping and restarting at ID 1 for client, and ID 2 for server, and incrementing sequentially by 2s as stated above.
+- The Requester MUST skip IDs still in use. 
+- The Responder MAY choose to ERROR[REJECT] any Stream ID if it considers the ID still in use. The Requester MAY retry on the next sequential Stream ID considered unused.
```
Comment (yschimke) [inline Protocol.md]: You should review the last section of the doc

> Receiving a Request frame on a Stream ID that is already in use MUST be ignored.

**Pull request merged**